src/Confluent.Kafka/Producer.cs

```diff
@@ -39,30 +63,54 @@ public class Producer<TKey, TValue> : IProducer<TKey, TValue>
         private readonly Producer ownedClient;
         private readonly Handle handle;
         private Producer producer;
-        private ISerializer<TKey> keySerializer;
-        private ISerializer<TValue> valueSerializer;
+        private Serializer<TKey> keySerializer;
+        private Serializer<TValue> valueSerializer;
 
-        private void setAndValidateSerializers(ISerializer<TKey> keySerializer, ISerializer<TValue> valueSerializer)
+        private void setAndValidateSerializers(Serializer<TKey> keySerializer, Serializer<TValue> valueSerializer)
         {
             this.keySerializer = keySerializer;
             this.valueSerializer = valueSerializer;
 
-            if (keySerializer != null && keySerializer == valueSerializer)
+            if (keySerializer != null && (object)keySerializer == (object)valueSerializer)
             {
                 throw new ArgumentException("Key and value serializers must not be the same object.");
             }
 
             if (keySerializer == null)
             {
-                if (typeof(TKey) != typeof(Null))
+                if (typeof(TKey) == typeof(Null))
+                {
+                    this.keySerializer = Serializers.Null as Serializer<TKey>;
+                }
+                else if (typeof(TKey) == typeof(byte[]))
+                {
+                    this.keySerializer = Serializers.ByteArray as Serializer<TKey>;
+                }
+                else if (typeof(TKey) == typeof(string))
+                {
+                    this.keySerializer = Serializers.UTF8 as Serializer<TKey>;
+                }
+                else 
                 {
                     throw new ArgumentNullException("Key serializer must be specified.");
                 }
             }
 
             if (valueSerializer == null)
             {
-                if (typeof(TValue) != typeof(Null))
+                if (typeof(TValue) == typeof(Null))
+                {
+                    this.valueSerializer = Serializers.Null as Serializer<TValue>;
+                }
+                else if (typeof(TValue) == typeof(byte[]))
```

Comment:
should this be isinstanceof rather than typeof?

Reply:
nope